### PR TITLE
Do not return localhost as split location from Hive connector

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public String getPartitionName()
 
     private Optional<InternalHiveSplit> createInternalHiveSplit(LocatedFileStatus status, OptionalInt bucketNumber, boolean splittable)
     {
-        splittable = splittable && isSplittable(inputFormat, fileSystem, status.getPath());
+        splittable &= isSplittable(inputFormat, fileSystem, status.getPath());
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java]: This no longer short-circuits. `&&=` would... if it existed.
If your intention is to invoke `isSplittable` unconditionally, add a comment in the code.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java [hunk 2]
```diff
@@ -232,22 +232,17 @@ private static boolean allBlocksHaveAddress(Collection<InternalHiveBlock> blocks
...
-
-        return effectivePredicate.getDomains().get().entrySet().stream()
-                .filter(entry -> isPathColumnHandle(entry.getKey()))
-                .findFirst()
-                .map(Map.Entry::getValue);
+        return effectivePredicate.getDomains()
+                .flatMap(domains -> domains.entrySet().stream()
+                        .filter(entry -> isPathColumnHandle(entry.getKey()))
+                        .findFirst()
+                        .map(Map.Entry::getValue));
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java]: I'd move `.map(Map.Entry::getValue)` before `.findFirst()`

File: presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java (hunk 1 above)
Comment (electrum) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java]: Good point. Reverted

File: presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java (hunk 2 above)
Comment (electrum) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java]: Done

**Pull request merged**